contracts/escrows/CountdownGriefingEscrow.sol

```diff
@@ -0,0 +1,357 @@
+        if (stakeAmount != uint256(0))
+            require(stakeAmount == uint256(uint128(stakeAmount)), "stakeAmount is too large");
+            _data.stakeAmount = uint128(stakeAmount);
+
+        // set countdown length
+        Countdown._setLength(escrowCountdown);
+
+        // set metadata
+        if (metadata.length != 0)
+            EventMetadata._setMetadata(metadata);
+
+        // set agreementParams
+        if (agreementParams.length != 0) {
+            (
+                uint120 ratio,
+                uint8 ratioType,
+                uint128 agreementCountdown
+            ) = abi.decode(agreementParams, (uint120, uint8, uint128));
+
+            _data.agreementParams = AgreementParams(ratio, ratioType, agreementCountdown);
+        }
+
+        // emit event
+        emit Initialized(buyer, seller, operator, paymentAmount, stakeAmount, escrowCountdown, metadata, agreementParams);
+    }
+
+    /// @notice Emit metadata event
+    /// @dev Access Control: seller OR buyer OR operator
+    ///      State Machine: always
+    function setMetadata(bytes memory metadata) public {
+        // restrict access
+        require(isSeller(msg.sender) || isBuyer(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or buyer or active operator");
+
+        // update metadata
+        EventMetadata._setMetadata(metadata);
+    }
+
+    /// @notice Deposit Stake
+    ///          - if seller not already set, make msg.sender the seller
+    ///          - if buyer already deposited the payment, finalize the escrow
+    /// @dev Access Control: buyer OR operator
+    ///      State Machine: before finalize() OR before cancel()
+    function depositStake() public {
+        // restrict access control
+        // set msg.sender as seller if not already set
+        if (!isSeller(msg.sender) && !Operated.isActiveOperator(msg.sender)) {
+            require(_data.seller == address(0), "only seller or active operator");
+            _data.seller = msg.sender;
+        }
+        // restrict state machine
+        require(!isFinalized() && !isCancelled(), "only before finalize or cancel");
+
+        // Add the stake amount
+        if (uint256(_data.stakeAmount) != uint256(0))
+            Staking._addStake(_data.seller, msg.sender, uint256(0), uint256(_data.stakeAmount));
+
+        // If payment is deposited, finalize the escrow
+        if (isPaymentDeposited())
+            _data.status = EscrowStatus.isDeposited;
+            finalize();
+
+        // emit event
+        emit StakeDeposited(_data.seller, uint256(_data.stakeAmount));
+    }
+
+    /// @notice Deposit Payment
+    ///          - if buyer not already set, make msg.sender the buyer
+    ///          - if seller already deposited the stake, start the finalization countdown
+    /// @dev Access Control: buyer OR operator
+    ///      State Machine: before finalize() OR before cancel()
+    function depositPayment() public {
```

Comment:
The buyer could call depositPayment multiple times and get the contract into a weird state.

Reply:
same as depositStake()

Reply:
fixed in 50df9d5